packages/contracts-bedrock/VERSIONING.md

```diff
@@ -0,0 +1,128 @@
+
+> - `patch` releases are to be used only for changes that do NOT modify contract bytecode (such as updating comments).
+> - `minor` releases are to be used for changes that modify bytecode OR changes that expand the contract ABI provided that these changes do NOT break the existing interface.
+> - `major` releases are to be used for changes that break the existing contract interface OR changes that modify the security model of a contract.
+>
+> Bumping the patch version does change the bytecode, so another exception is carved out for this.
+> In other words, changing comments increments the patch version, which changes bytecode. This bytecode
+> change implies a minor version increment is needed, but because it's just a version change, only a
+> patch increment should be used.
+
+## Individual Contract Versioning
+
+Versioning for individual contracts works as follows:
+
+- A contract is only `X.Y.Z` on `develop` if it has been governance approved. If it's `X.Y.Z` before that, it must be on a branch. More on this below.
+- For contracts undergoing development, a `-beta.n` identifier must be appended to the version number.
+- For contracts in a release candidate state, an `-rc.n` identifier must be appended to the version number.
+- For contracts with feature-specific changes, a `+feature-name` identifier must be appended to the version number. See the [Smart Contract Feature Development](https://github.com/ethereum-optimism/design-docs/blob/main/smart-contract-feature-development.md) design document to learn more.
+- When making changes to a contract, always bump to the lowest possible version based on the specific change you are making. We do not want to e.g. optimistically bump to a major version, because protocol development sequencing may change unexpectedly. Use these examples to know how to bump the version:
+  - Example 1: A contract is currently on `1.2.3`.
+    - We don't yet know when the next release of this contract will be. However, you are simply fixing typos in comments so you bump the version to `1.2.4-beta.1`.
+    - The next PR made to that same contract clarifies some comments, so it bumps the version to `1.2.4-beta.2`.
+    - The last PR introduces a breaking change, which bumps the version from `1.2.4-beta.2` to `2.0.0-beta.1`. A `1.2.4-rc.1` and `1.2.4` version both never exist.
+  - Example 2: A contract is currently on `2.4.7`.
+    - We know the next release of this contract will be a breaking change. Regardless, as you start development by fixing typos in comments, bump the version to `2.4.8-beta.1`. This is because we may end up putting out a release before the breaking change is added.
+    - Once you start working on the breaking change, bump the version to `3.0.0-beta.1`.
+    - Continue to bump the beta version as you make changes. When the contract is ready for release, bump the version to `3.0.0-rc.1`.
+- New contracts start at `0.y.z` and only become `1.0.0` when they are ready for production.
```

Comment:
This was what we approved in the design review, but I would actually suggest the following change—it is simpler, and avoids confusion and questions around whether `0.y.z` versions are production ready. I think this is important because a lot of tools in the industry are production tools that are versioned below 1.0.0 (solidity, vyper, foundry, etc). Are there any objections to this?
```suggestion
- New contracts start at `1.0.0-beta.1`, increment the `-beta.n` counter during development, and become `1.0.0` when they are ready for production.
```

Reply:
I like this change, it reduces the overhead of developing new contracts especially when using smaller PRs.

Reply:
I've applied my own suggestion, but will keep this unresolved in case other reviewers have different thoughts